### PR TITLE
[DOCS] Update security doc ML requirements

### PR DESCRIPTION
**Pull request by benironside**
Addresses #1256  - This [security doc](https://www.elastic.co/guide/en/security/current/ml-requirements.html#ml-requirements) says that ML has to be enabled on _all_ master-eligible and coordinating ES nodes, but the [ML doc](https://www.elastic.co/guide/en/machine-learning/current/setup.html#ml-nodes) says "To use machine learning features, there must be _at least one_ machine learning node in your cluster."

Planning to backport this to versions >= 7.9, as suggested by Ryland in the issue. The link above that Ryland requested we use to replace the previous link does not exist on the main branch, 7.16, or 8.0, so I replaced it with https://www.elastic.co/guide/en/machine-learning/current/setup.html (same link without the anchor link tag).

Preview [here](https://security-docs_1281.docs-preview.app.elstc.co/guide/en/security/master/ml-requirements.html).

### CODE REVIEW

File: docs/getting-started/ml-req.asciidoc [hunk 1]
```diff
@@ -4,7 +4,6 @@
 To run and create {ml} jobs and rules, you need all of these:
 
 * The *https://www.elastic.co/subscriptions[appropriate license]*
-* {ml-cap} enabled on all master-eligible and coordinating {es} nodes (see
-{ref}/modules-node.html#ml-node[{ml-cap} node])
+* There must be at least one {ml} node in your cluster (see {ml-docs}/setup.html[{ml-cap} node])
```
Comment (jmikell821) [inline docs/getting-started/ml-req.asciidoc:7]: Hi @benironside @joepeeples - confused if we are linking to the top of the doc here? The title says `"Set up machine learning features"`, not the `"Machine learning node"` link text mentioned.

Comment (joepeeples) [reply docs/getting-started/ml-req.asciidoc:7]: Ah yes, we updated the link but forgot to update the link text. This should do it (looks like the title of that page has stayed the same over the previous versions).
```suggestion
* There must be at least one {ml} node in your cluster (see {ml-docs}/setup.html[Set up {ml} features])
```

**Review by rylnd — APPROVED** [review]
LGTM

**Review by jmikell821 — APPROVED** [review]
LGTM. @joepeeples if you could walk Ben through the manual backport process since this needs to be backported all the way to `7.9.0`. Mergify only goes back to `7.14.0` so it may just be easier to do a manual backport.

**Review by jmikell821 — APPROVED** [review]

**Review by joepeeples — APPROVED** [review]

**Review by nastasha-solomon — APPROVED** [review]

**Pull request merged**